src/elf.rs

```diff
@@ -1,18 +1,25 @@
+use crate::error::{GetSymbolsError, Result};
 use compact_symbol_table::CompactSymbolTable;
 use goblin::elf;
 use object::{ElfFile, Object, Uuid};
 use std::cmp;
-
 const UUID_SIZE: usize = 16;
 const PAGE_SIZE: usize = 4096;
 
-pub fn get_compact_symbol_table(buffer: &[u8], breakpad_id: &str) -> Option<CompactSymbolTable> {
-    let elf_file = ElfFile::parse(buffer).ok()?;
-    let elf_id = get_elf_id(&elf_file, buffer)?;
-    if format!("{:X}0", elf_id.simple()) != breakpad_id {
-        return None;
+pub fn get_compact_symbol_table(buffer: &[u8], breakpad_id: &str) -> Result<CompactSymbolTable> {
+    let elf_file = ElfFile::parse(buffer).or(Err(GetSymbolsError::InvalidInputError(
+        "Could not parse ELF header",
+    )))?;
+    let elf_id = get_elf_id(&elf_file, buffer)
+        .ok_or(GetSymbolsError::InvalidInputError("id cannot be read"))?;
```

Comment:
And use `.ok_or_else` in the places where you're using `.ok`, for the same reason.

Reply:
Oh, here I meant to say "use `.ok_or_else` in the places where you're using `.ok_or`.